product_customer_code/models/product_customer_code.py

```diff
@@ -0,0 +1,35 @@
+# coding: utf-8
+
+from openerp import api, fields, models
+
+
+class ProductCustomerCode(models.Model):
+
+    _name = "product.customer.code"
+    _description = "Add manies Code of Customer's"
+
+    _rec_name = 'product_code'
+
+    product_code = fields.Char('Customer Product Code', required=True,
+                               index=True, help="""This customer's product
+                               code will be used when searching into a request
+                               for quotation.""")
+    product_name = fields.Char('Customer Product Name', help="""This customer's
+                               product name will be used when searching into a
+                               request for quotation.""")
+    product_id = fields.Many2one('product.product', 'Product', required=True,
+                                 help='Product Identification')
+    partner_id = fields.Many2one('res.partner', 'Customer', required=True,
+                                 index=True, help='Partner Reference')
+    company_id = fields.Many2one('res.company', 'Company', default=lambda self:
+                                 self._get_user_default_company())
```

Comment:
What about use `default=lambda self: self.env['res.company']._company_default_get('product.customer.code')`?

Similar to [event - default company](https://github.com/odoo/odoo/blob/27a47eb3c3d4b79d8fffb6df66f107a993ddf805/addons/event/event.py#L176)